azure-test/tests/azure_virtual_network_gateway/variables.tf

```diff
@@ -0,0 +1,102 @@
+variable "resource_name" {
+  type        = string
+  default     = "turbot-test-20200125-create-update"
+  description = "Name of the resource used throughout the test."
+}
+
+variable "azure_environment" {
+  type        = string
+  default     = "public"
+  description = "Azure environment used for the test."
+}
+
+variable "azure_subscription" {
+  type        = string
+  default     = "3510ae4d-530b-497d-8f30-53b9616fc6c1"
+  description = "Azure subscription used for the test."
+}
+
+provider "azurerm" {
+  # Cannot be passed as a variable
+  version         = "=1.36.0"
+  environment     = var.azure_environment
+  subscription_id = var.azure_subscription
+}
+
+data "azurerm_client_config" "current" {}
+
+data "null_data_source" "resource" {
+  inputs = {
+    scope = "azure:///subscriptions/${data.azurerm_client_config.current.subscription_id}"
+  }
+}
+
+resource "azurerm_resource_group" "named_test_resource" {
+  name     = var.resource_name
+  location = "East US"
+}
+
+resource "azurerm_virtual_network" "named_test_resource" {
+  depends_on = [azurerm_resource_group.named_test_resource]
+  name                = var.resource_name
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.named_test_resource.location
+  resource_group_name = azurerm_resource_group.named_test_resource.name
+}
+
+resource "azurerm_subnet" "named_test_resource" {
+  depends_on = [azurerm_virtual_network.named_test_resource]
+  name                 = "GatewaySubnet"
+  resource_group_name  = azurerm_resource_group.named_test_resource.name
+  virtual_network_name = azurerm_virtual_network.named_test_resource.name
+  address_prefix       = "10.0.1.0/24"
+}
+
+resource "azurerm_public_ip" "named_test_resource" {
+  depends_on = [azurerm_subnet.named_test_resource]
+  name                = var.resource_name
+  location            = azurerm_resource_group.named_test_resource.location
+  resource_group_name = azurerm_resource_group.named_test_resource.name
+  allocation_method   = "Dynamic"
+}
+
+locals {
+  path = "${path.cwd}/info.json"
+}
+
+resource "null_resource" "named_test_resource" {
+  depends_on = [azurerm_public_ip.named_test_resource]
+  provisioner "local-exec" {
+    command = "az network vnet-gateway create --gateway-type Vpn --location eastus --name ${var.resource_name} --no-wait --public-ip-addresses ${var.resource_name} --resource-group ${var.resource_name} --vnet ${var.resource_name}"
```

Comment:
@bigdatasourav Since you are creating the resource using CLI, resources are not deleted during teardown process, and the teardown process also failed due to dependencies
```
Error: Error deleting Public IP "turbottest93951" (Resource Group "turbottest93951"): network.PublicIPAddressesClient#Delete: Failure sending request: StatusCode=400 -- Original Error: Code="PublicIPAddressCannotBeDeleted" Message="Public IP address /subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/resourceGroups/turbottest93951/providers/Microsoft.Network/publicIPAddresses/turbottest93951 can not be deleted since it is still allocated to resource /subscriptions/d7245080-b4ae-4fe5-b6fa-2e71b3dae6c8/resourceGroups/turbottest93951/providers/Microsoft.Network/virtualNetworkGateways/turbottest93951/ipConfigurations/vnetGatewayConfig0. In order to delete the public IP, disassociate/detach the Public IP address from the resource.  To learn how to do this, see aka.ms/deletepublicip." Details=[]
```
Can we add `posttest` to delete the resources?

Reply:
updated